pkgs/by-name/ca/caido/package.nix

```
@@ -1,40 +1,92 @@
-{ lib
-, fetchurl
-, appimageTools
-, makeWrapper
+{
+  lib,
+  stdenv,
+  fetchurl,
+  appimageTools,
+  makeWrapper,
+  autoPatchelfHook,
+  libgcc,
+  appVariants ? [ ],
 }:
 
 let
   pname = "caido";
-  version = "0.33.0";
-  src = fetchurl {
+  appVariantList = [
+    "cli"
+    "desktop"
+  ];
+  version = "0.39.0";
+  cli = fetchurl {
+    url = "https://storage.googleapis.com/caido-releases/v${version}/caido-cli-v${version}-linux-x86_64.tar.gz";
+    hash = "sha256-I8UF2rzIKfpcrxyvDa4AReWDIHOKTCj3ERaWhG1xGG0=";
+  };
+  desktop = fetchurl {
     url = "https://storage.googleapis.com/caido-releases/v${version}/caido-desktop-v${version}-linux-x86_64.AppImage";
-    hash = "sha256-MUQ1tVcIpLrC2RKsWDqv8MBGaHfh56OxIC/ARArQjiU=";
+    hash = "sha256-KYQck2+YYPLJN3L6qchacjyVyyXR3nmJDTX5GPB4WvI=";
+  };
+  appimageContents = appimageTools.extractType2 {
+    inherit pname version;
+    src = desktop;
+  };
+
+  wrappedDesktop = appimageTools.wrapType2 {
+    src = desktop;
+    inherit pname version;
+
+    extraPkgs = pkgs: [ pkgs.libthai ];
+
+    extraInstallCommands = ''
+      install -m 444 -D ${appimageContents}/caido.desktop -t $out/share/applications
+      install -m 444 -D ${appimageContents}/caido.png \
+        $out/share/icons/hicolor/512x512/apps/caido.png
+      source "${makeWrapper}/nix-support/setup-hook"
+      wrapProgram $out/bin/${pname} \
```

```suggestion
      wrapProgram $out/bin/caido \
```
see the pname overuse tracking issue